core/modules/link/link.module

```diff
@@ -75,6 +77,18 @@ function link_field_instance_settings_form($field, $instance) {
     '#description' => t('If checked, the URL field is optional and submitting a title alone will be acceptable. If the URL is omitted, the title will be displayed as plain text.'),
   );
 
+  $form['type'] = array(
+    '#type' => 'radios',
+    '#title' => t('URL type'),
+    '#default_value' => $instance['settings']['type'],
+    '#options' => array(
+      LINK_INTERNAL => t('Internal only'),
+      LINK_EXTERNAL => t('External only'),
+      LINK_BOTH => t('Both (internal and external)'),
+    ),
+    '#description' => t('The type of URLs this field will allow. Internal URLs point to pages on this website; an autocomplete displays matching pages to select from. External URLs point to other websites; no autocomplete is used.'),
+  );
```

Comment:
The "The type of URLs this field will allow." bit does not provide any real value - we can move the word "allow"/"allowed" to be part of the radio set label, which will have the same effect.

Then can we please add individual, shorter descriptions to the various options, instead of a single lengthy one?:
```suggestion
  );
  $form['type'][LINK_INTERNAL]['#description'] = t('Internal URLs point to pages on this website; an autocomplete displays matching pages to select from.');
  $form['type'][LINK_EXTERNAL]['#description'] = t('External URLs point to other websites; no autocomplete is used.');
```